src/renderer/src/routes/modals/Search/index.tsx

```diff
@@ -0,0 +1,206 @@
+import { Separator } from '@renderer/components/Separator'
+import { SynonymsHelper } from '@renderer/helpers/SynonymsHelper'
+import { ToastHelper } from '@renderer/helpers/ToastHelper'
+import { useActions } from '@renderer/hooks/useActions'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
+import { CenterModalLayout } from '@renderer/layouts/CenterModal'
+import { search } from 'fast-fuzzy'
+import { debounce } from 'lodash'
+import { removeStopwords } from 'stopword'
+import { match } from 'ts-pattern'
+import winkWebModel from 'wink-eng-lite-web-model'
+import WinkNLP from 'wink-nlp'
+
+import { functionByActionId } from './functionByActionId'
+
+const nlp = WinkNLP(winkWebModel)
+
+type TActionData = {
+  isSearching: boolean
+  foundActions?: TSearchAction[]
+  search: string
+}
+
+type TSearchAction = {
+  label: string
+  verbs: string[]
+  nonVerbs: string[]
+  id: string
+}
+
+export const SearchModal = () => {
+  const { modalEraseWrapper } = useModalNavigate()
+  const { t } = useTranslation('modals', { keyPrefix: 'search' })
+  const { t: tCommon } = useTranslation('common')
+  const modalActions = useModalNavigate()
+  const pageNavigate = useNavigate()
+
+  const { actionData, setData } = useActions<TActionData>({ isSearching: false, search: '', foundActions: undefined })
+
+  const searchActions = useMemo<TSearchAction[]>(() => {
+    const actions = tCommon('searchActions', { returnObjects: true })
+    return actions
+  }, [tCommon])
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleSearch = useCallback(
+    debounce(async (text: string) => {
+      const doc = nlp.readDoc(text.toLowerCase())
+      const tokens = doc.tokens().filter(t => t.out(nlp.its.type) === 'word')
+      const verbs = tokens.filter(token => token.out(nlp.its.pos) === 'VERB').out(nlp.its.lemma as any) as string[]
+      const nonVerbs = removeStopwords(
+        tokens.filter(token => token.out(nlp.its.pos) !== 'VERB').out(nlp.its.lemma as any) as string[]
+      )
+
+      try {
+        if (verbs.length === 0 && nonVerbs.length === 0) {
+          setData({ foundActions: [] })
+          return
+        }
+
+        const foundActions: TSearchAction[] = []
+
+        for (const action of searchActions) {
+          const allVerbsSynonyms = await SynonymsHelper.getAllSynonyms(action.verbs)
+          const allNonVerbsSynonyms = await SynonymsHelper.getAllSynonyms(action.nonVerbs)
+
+          const allVerbs = [...action.verbs, ...allVerbsSynonyms]
+          const allNonVerbs = [...action.nonVerbs, ...allNonVerbsSynonyms]
+
+          const hasVerb = verbs.some(value => search(value, allVerbs).length > 0)
+          const hasNonVerb = nonVerbs.some(item => search(item, allNonVerbs).length > 0)
+
+          if (verbs.length > 0 && nonVerbs.length > 0) {
+            if (hasVerb && hasNonVerb) {
+              foundActions.push(action)
+            }
+          } else if (nonVerbs.length > 0 && hasNonVerb) {
```

Comment:
What do you think?

```typescript
if (verbs.length > 0 && nonVerbs.length > 0 && hasVerb && hasNonVerb) { ... }
```

Reply:
I tried this way, but the result also includes some searches that are not so assertive